### PR TITLE
YD-159 Add creationTime to goal DTO

### PR DESCRIPTION
**Pull request by aristotelos**
The property was already added earlier for inactivity support.
One question: Should we also refactor Message.creationTime to ZonedDateTime? That would be more consistent. Or the other way around, expose a Date here?

### CODE REVIEW

File: appservice/src/intTest/groovy/nu/yona/server/EditGoalsTest.groovy [hunk 1]
```diff
@@ -51,10 +51,12 @@ class EditGoalsTest extends AbstractAppServiceIntegrationTest
 		gamblingGoals.size() == 1
 		gamblingGoals[0]."@type" == "BudgetGoal"
 		!gamblingGoals[0]._links.edit //mandatory goal
+		gamblingGoals[0].creationTime =~ /\d{4}\-\d{2}\-\d{2}T\d{2}\:\d{2}\:\d{2}\.\d{3}\+\d{2}\:\d{2}/
```
Comment (Bert-R) [inline appservice/src/intTest/groovy/nu/yona/server/EditGoalsTest.groovy]: Can't you add an assert that the creation time is equal to the time the user was created?

File: appservice/src/intTest/groovy/nu/yona/server/EditGoalsTest.groovy [hunk 2]
```diff
@@ -51,10 +51,12 @@ class EditGoalsTest extends AbstractAppServiceIntegrationTest
...
 		def newsGoals = filterGoals(response, NEWS_ACT_CAT_URL)
 		newsGoals.size() == 1
 		newsGoals[0]."@type" == "BudgetGoal"
 		newsGoals[0]._links.edit.href
+		newsGoals[0].creationTime =~ /\d{4}\-\d{2}\-\d{2}T\d{2}\:\d{2}\:\d{2}\.\d{3}\+\d{2}\:\d{2}/
```
Comment (Bert-R) [inline appservice/src/intTest/groovy/nu/yona/server/EditGoalsTest.groovy]: Similar here, but for the time the goal was created.

File: appservice/src/main/resources/static/swagger/swagger-spec.yaml [hunk 3]
```diff
@@ -1201,9 +1201,15 @@ definitions:
           - $ref: '#/definitions/SelfLink'
           - $ref: '#/definitions/EditLink'
           - $ref: '#/definitions/ActivityCategoryLink'
+      creationTime:
+        type: string
+        format: dateTime
+        description: The creation time of this goal
+        example: 2016-04-26T20:08:55.365+02:00
```
Comment (Bert-R) [inline appservice/src/main/resources/static/swagger/swagger-spec.yaml]: Other examples use this time stamp sample format: ``2016-02-31T21:28:02.290+0000`` (without the colon in the middle). Pleas use that here too.

Comment (Bert-R) [inline appservice/src/main/resources/static/swagger/swagger-spec.yaml]: I just now see in the test that the serialization format also differs. Please update that too.

File: core/src/main/java/nu/yona/server/goals/service/GoalDTO.java [hunk 4]
```diff
@@ -45,6 +48,11 @@ public UUID getID()
 		return id;
 	}
 
+	public ZonedDateTime getCreationTime()
```
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/goals/service/GoalDTO.java]: To change the serialization format, you will need to add an annotation. In ``AppActivityDTO``, I used the following for deserialization:
```
@JsonFormat(pattern = **"yyyy-MM-dd'T'HH:mm:ss.SSSZ")
```

File: appservice/src/main/resources/static/swagger/swagger-spec.yaml (hunk 3 above)
Comment (aristotelos) [inline appservice/src/main/resources/static/swagger/swagger-spec.yaml]: How should I do that? Is it a Jackson setting? The type is different: the other places serialize `Date`, this place serializes `ZonedDateTime`...

**Comment by Bert-R** [comment]
> One question: Should we also refactor Message.creationTime to ZonedDateTime? That would be more consistent. Or the other way around, expose a Date here?

The direction we evolve in is to use ``ZonedDateTime``, so using that here is appropriate. We need to get rid of all occurrences of ``Date`` some day, not necessarily now. For as long as the external API remains te same, we can stick with ``Date`` till we find time.

**Comment by Bert-R** [comment]
I have added a few comments related to the date format and checking.

File: appservice/src/main/resources/static/swagger/swagger-spec.yaml (hunk 3 above)
Comment (Bert-R) [inline appservice/src/main/resources/static/swagger/swagger-spec.yaml]: See comment on ``GoalDTO``.

**Pull request merged**